x-pack/plugins/observability_solution/dataset_quality/server/routes/data_streams/get_data_stream_details/index.ts

```diff
@@ -82,17 +81,17 @@ export async function getDataStreamDetails({
     );
 
     const whenSizeStatsNotAvailable = NaN; // This will indicate size cannot be calculated
-    const avgDocSizeInBytes = sizeStatsAvailable
-      ? hasAccessToDataStream && dataStreamSummaryStats.docsCount > 0
-        ? await getAvgDocSizeInBytes(esClient, dataStream)
-        : 0
-      : whenSizeStatsNotAvailable;
+    const avgDocSizeInBytes = isServerless
```

Comment:
nit: never found chained ternary operations well readable.

Reply:
I agree on this one, however this will be gone/improved with https://github.com/elastic/logs-dev/issues/179